### PR TITLE
Add hide card feature to RewardsCardPlugin

### PR DESCRIPTION
**Pull request by samholmes**
### CHANGELOG

- Added: Delete card feature to Visa Card Program

### Dependencies

 none

### Requirements

If you have made **any** visual changes to the GUI. Make sure you have:

- [ ] Tested on iOS device
- [ ] Tested on Android device
- [ ] Tested on small-screen device (iPod Touch)
- [ ] Tested on large-screen device (tablet)

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204616457260595

Simulator Screen Shot - iPhone 14 - 2023-05-24 at 12 04 47

Simulator Screen Shot - iPhone 14 - 2023-05-24 at 12 04 44

Simulator Screen Shot - iPhone 14 - 2023-05-24 at 12 04 38

Simulator Screen Shot - iPhone 14 - 2023-05-24 at 12 04 34

Simulator Screen Shot - iPhone 14 - 2023-05-24 at 12 04 28

Simulator Screen Shot - iPhone 14 - 2023-05-24 at 13 28 02

### CODE REVIEW

File: src/plugins/gui/providers/ioniaProvider.ts [hunk 1]
```diff
@@ -223,6 +228,8 @@ export const makeIoniaProvider: FiatProviderFactory<IoniaMethods> = {
     // State:
     //
 
+    let hiddenCardIds: number[] = asStoreHiddenCards(await store.getItem(STORE_HIDDEN_CARDS_KEY).catch(_ => undefined))
+    console.log('!@!', { hiddenCardIds })
```
Comment (peachbits) [inline src/plugins/gui/providers/ioniaProvider.ts:232]: Could be useful for debugging to have these logged but with a more useful message

File: src/plugins/gui/RewardsCardPlugin.tsx [hunk 2]
```diff
@@ -42,13 +42,13 @@ export const makeRewardsCardPlugin: FiatPluginFactory = async params => {
...
       // Expires 6 calendar months from the creation date
       const expirationDate = new Date(card.CreatedDate.valueOf())
       expirationDate.setMonth(card.CreatedDate.getMonth() + 6)
 
       return {
-        id: card.Id.toString(),
+        id: card.Id,
         // Expires in 180 days (6 months) from creation date
         expiration: expirationDate,
         url: card.CardNumber
```
Comment (peachbits) [inline src/plugins/gui/RewardsCardPlugin.tsx:54]: This is actually a comment for line 58 I think the old cards that expire first should be at the top of the list.

File: src/plugins/gui/providers/ioniaProvider.ts [hunk 3]
```diff
@@ -391,12 +398,25 @@ export const makeIoniaProvider: FiatProviderFactory<IoniaMethods> = {
...
+          await store.setItem(STORE_HIDDEN_CARDS_KEY, wasStoreHiddenCards(hiddenCardIds))
+        },
         async getGiftCards() {
           const giftCardsResponse = await fetchGetGiftCards({
             headers: userAuthenticatedFetchOptions.headers
           })
           const { Data: giftCards } = giftCardsResponse
-          return giftCards
+
+          const out: { cards: GiftCard[]; archivedCards: GiftCard[] } = { cards: [], archivedCards: [] }
```
Comment (peachbits) [inline src/plugins/gui/providers/ioniaProvider.ts:413]: archivedCards are defined but unused

File: src/plugins/gui/scenes/RewardsCardDashboardScene.tsx [hunk 4]
```diff
@@ -100,15 +99,17 @@ const CardListContainer = styled(View)<{ bottomSpace: number }>(props => ({
 }))
 
 const CardListItem = styled(View)(props => ({
-  marginLeft: props.theme.rem(1)
+  backgroundColor: props.theme.modal,
```
Comment (peachbits) [inline src/plugins/gui/scenes/RewardsCardDashboardScene.tsx:102]: Would be nice if the cards adhered to industry standard CR80 ratio (3.375 inches wide by 2.125 inches high)

**Review by peachbits — CHANGES_REQUESTED** [review]

File: src/plugins/gui/providers/ioniaProvider.ts (hunk 1 above)
Comment (samholmes) [reply src/plugins/gui/providers/ioniaProvider.ts:232]: Not intended to be there.

File: src/plugins/gui/providers/ioniaProvider.ts (hunk 3 above)
Comment (samholmes) [reply src/plugins/gui/providers/ioniaProvider.ts:413]: Ah, yeah, I was thinking about using them in the UI later. Like we could show text-button "Show Archived Cards" if we see the length > 0 for archivedCards

File: src/plugins/gui/RewardsCardPlugin.tsx [hunk 5]
```diff
@@ -106,19 +105,26 @@ export const makeRewardsCardPlugin: FiatPluginFactory = async params => {
...
         bridge={bridge}
         buttons={{
-          keep: { label: lstrings.string_keep, type: 'primary' },
-          delete: { label: lstrings.string_delete, type: 'escape' }
+          delete: { label: lstrings.string_delete, type: 'secondary' },
+          keep: { label: lstrings.string_keep, type: 'escape' }
         }}
         title={lstrings.rewards_card_delete_modal_title}
-        message={sprintf(lstrings.rewards_card_delete_modal_message_s, item.expiration)}
+        message={sprintf(lstrings.rewards_card_delete_modal_message_s, card.expiration)}
```
Comment (peachbits) [inline src/plugins/gui/RewardsCardPlugin.tsx:117]: While still imperfect, the text could be a bit clearer like "Are you sure you want to delete the Visa Card with the following expiration?”

Or even "Are you sure you want to delete the Visa Card that expires %s?”

Either way, format the expiration with toLocaleString() as you did in the card itself

**Review by peachbits — CHANGES_REQUESTED** [review]

**Review by peachbits — APPROVED** [review]

**Pull request merged**